pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -175,6 +175,21 @@ func (s *requested) ExecuteInbound(msg *stateMachineMsg, thid string, ctx *conte
 			return nil, nil, nil, fmt.Errorf("JSON unmarshalling of invitation: %w", err)
 		}
 
+		// Invitation received
+		// TODO: For now, PID can be set only by the introduce protocol.
+		//  if Thread.PID is set, we need to notify the introduce protocol about this action.
+		//  This logic should be changed with some generic solution. What if another protocol sets the PID?
+		//  How to determine which protocol provides PID? Which action we should take?
+		if invitation.Thread != nil && invitation.Thread.PID != "" {
+			if err = ctx.loadIntroduceNotifier(); err != nil {
```

Comment:
The notifier hook is being modified in the middle of the state machine?

Reply:
Done!